### PR TITLE
Fix restore size type in snapshot content

### PR DESCRIPTION
**Pull request by jingxu97**
Change the restore size to int64 for snapshot content. Also add the code
to update the restore size for snapshot content.

### CODE REVIEW

File: pkg/controller/snapshot_controller.go [hunk 1]
```diff
@@ -637,6 +641,28 @@ func (ctrl *csiSnapshotController) bindandUpdateVolumeSnapshot(snapshotContent *
 	return snapshotCopy, nil
 }
 
+// updateSnapshotContentSize update the restore size for snapshot content
+func (ctrl *csiSnapshotController) updateSnapshotContentSize(content *crdv1.VolumeSnapshotContent, size int64) error {
+	if content.Spec.VolumeSnapshotSource.CSI == nil || size == 0 {
```
Comment (xing-yang) [inline pkg/controller/snapshot_controller.go]: size <= 0

Comment (jingxu97) [reply pkg/controller/snapshot_controller.go]: done

File: pkg/controller/snapshot_controller.go [hunk 2]
```diff
@@ -637,6 +641,28 @@ func (ctrl *csiSnapshotController) bindandUpdateVolumeSnapshot(snapshotContent *
...
+		return nil
+	}
+	if content.Spec.VolumeSnapshotSource.CSI.RestoreSize != nil && *content.Spec.VolumeSnapshotSource.CSI.RestoreSize > 0 {
```
[the remark this answers is not in the document]
Comment (jingxu97) [reply pkg/controller/snapshot_controller.go]: actually no, changed it to check whether the updated size is equal to the old one, 
*content.Spec.VolumeSnapshotSource.CSI.RestoreSize == size

**Pull request merged**